dask/array/rechunk.py

```diff
@@ -214,7 +213,16 @@ def rechunk(x, chunks, threshold=None,
     >>> y = x.rechunk({0: -1, 1: 'auto'}, block_size_limit=1e8)
     """
     if isinstance(chunks, dict):
-        chunks = dict(chunks)
+        new = {}
+        for k, v in chunks.items():
+            if not isinstance(k, int) or k >= x.ndim or k <= -x.ndim:
+                msg = ("Invalid dimension %s. Should be an integer from 0 to %d"
+                       % (k, x.ndim))
+                raise ValueError(msg)
+            if k < 0:
+                k += x.ndim
```

Comment:
Thoughts on raising here if that would clobber a key in `new`? e.g. the user wrote `x.rechunk({2: 1, -1: 1})` for a 2-D array.

Reply:
Meh.  I say we wait until this actually happens.